server/src/main/java/org/opensearch/common/xcontent/XContentHelper.java

```diff
@@ -68,8 +68,13 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
             if (compressedInput.markSupported() == false) {
                 compressedInput = new BufferedInputStream(compressedInput);
             }
-            final XContentType contentType = XContentFactory.xContentType(compressedInput);
-            return XContentFactory.xContent(contentType).createParser(xContentRegistry, deprecationHandler, compressedInput);
+            try {
```

Comment:
Shouldn't this go where the compressedInput stream is created ?i.e. 
```
InputStream compressedInput = compressor.threadLocalInputStream(bytes.streamInput());
```

Reply:
Thank you. I included the above line of the assignment for `compressedInput` into the `try` block.